.github/actions/backup_and_restore/action.yml

```diff
@@ -0,0 +1,110 @@
+name: Backup_and_restore
+description: Backup PaaS postgres DB and restore to AKS
+
+inputs:
+  environment:
+    description: Environment to backup and restore
+    required: true
+  azure-credentials-paas:
+    required: true
+  azure-credentials-aks:
+    required: true
+  slack-webhook:
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Set KV environment variables
+      shell: bash
+      run: |
+        tf_vars_file=terraform/paas/workspace_variables/${{ inputs.environment }}.tfvars.json
+        echo "key_vault_name=$(jq -r '.key_vault_name' ${tf_vars_file})" >> $GITHUB_ENV
+        echo "key_vault_infra_secret_name=$(jq -r '.key_vault_infra_secret_name' ${tf_vars_file})" >> $GITHUB_ENV
+        echo "paas_space_name=$(jq -r '.paas_cf_space' ${tf_vars_file})" >> $GITHUB_ENV
+
+    - uses: azure/login@v1
+      with:
+        creds: ${{ inputs.azure-credentials-paas }}
+
+    - uses: DFE-Digital/keyvault-yaml-secret@v1
+      id: get-secrets
+      with:
+        keyvault: ${{ env.key_vault_name }}
+        secret: ${{ env.key_vault_infra_secret_name }}
+        key: CF_USER,CF_PASSWORD
+
+    - name: Setup cf cli
+      uses: DFE-Digital/github-actions/setup-cf-cli@master
+      with:
+        CF_USERNAME:   ${{ steps.get-secrets.outputs.CF_USER }}
+        CF_PASSWORD:   ${{ steps.get-secrets.outputs.CF_PASSWORD }}
+        CF_SPACE_NAME: ${{ env.paas_space_name }}
+        INSTALL_CONDUIT: true
+
+    - name: Setup postgres client
+      uses: DFE-Digital/github-actions/install-postgres-client@master
+
+    - name: Set environment variable
+      shell: bash
+      run: echo "backup_file_name=apply_${{ inputs.environment }}.sql.gz" >> $GITHUB_ENV
+
+    - name: Backup ${{ inputs.environment }} DB
+      shell: bash
+      run: |
+        cf conduit apply-postgres-${{ inputs.environment }} -- pg_dump -E utf8 --compress=1 --clean --if-exists --no-owner --verbose --no-password -f ${{ env.backup_file_name }}
+        cf logout
+        az logout
+
+#
+# Now restore to the equivalent AKS env database
+#
+
+    - uses: azure/login@v1
+      with:
+        creds: ${{ inputs.azure-credentials-aks }}
+
+    - name: Set up environment variables
+      shell: bash
+      run: |
+        case ${{ inputs.environment }} in
```

Comment:
Could we make it more generic by reading "cluster" in tfvars.json? And derive cluster_rg and cluster_name from it
app_name can be `apply-${{ inputs.environment }}`

This will make this action more reusable for other services